tests/helper/ssh.go

```diff
@@ -0,0 +1,290 @@
+}
+
+func setWinsize(f *os.File, w, h int) {
+	syscall.Syscall(syscall.SYS_IOCTL, f.Fd(), uintptr(syscall.TIOCSWINSZ),
+		uintptr(unsafe.Pointer(&struct{ h, w, x, y uint16 }{uint16(h), uint16(w), 0, 0})))
+}
+
+func TestSSHSftpServer(t *testing.T, user, pass, key string) string {
+	sshSrv := &sshd.Server{
+		SubsystemHandlers: map[string]sshd.SubsystemHandler{
+			"sftp": func(s sshd.Session) {
+				t.Log("Test SFTP server: handling session")
+				debugStream := os.Stderr
+				serverOptions := []sftp.ServerOption{
+					sftp.WithDebug(debugStream),
+				}
+				server, err := sftp.NewServer(s, serverOptions...)
+				if err != nil {
+					t.Log("sftp server init error:", err)
+					return
+				}
+				if err := server.Serve(); err == io.EOF {
+					server.Close()
+					fmt.Println("sftp client exited session.")
+				} else if err != nil {
+					fmt.Println("sftp server completed with error:", err)
+				}
+				t.Log("Test SFTP server completed handling session")
+			},
+		},
+	}
+	return TestSSHServer(t, sshSrv, user, pass, key)
+}
+
+func RunCmdPtySSH(addr, username, password, cmd string) ([]byte, error) {
+	cfg := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(), // #nosec G106 , tests need to be simple
+	}
+
+	conn, err := ssh.Dial("tcp", addr, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to connect to %s: %v", addr, err)
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to create session: %v", err)
+	}
+	defer session.Close()
+
+	// Set up terminal modes
+	modes := ssh.TerminalModes{
+		ssh.ECHO:          0,     // disable echoing
+		ssh.TTY_OP_ISPEED: 14400, // input speed = 14.4kbaud
+		ssh.TTY_OP_OSPEED: 14400, // output speed = 14.4kbaud
+	}
+	// Request pseudo terminal
+	if err := session.RequestPty("xterm", 40, 80, modes); err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to request PTY: %v", err)
+	}
+
+	// Set up standard input/output
+	stdin, err := session.StdinPipe()
+	if err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to get session stdin: %v", err)
+	}
+
+	stdout, err := session.StdoutPipe()
+	if err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to get session stdout: %v", err)
+	}
+
+	// Start remote shell
+	if err := session.Shell(); err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to request shell: %v", err)
+	}
+
+	// Send command
+	if _, err = io.WriteString(stdin, fmt.Sprintf("%s\n", cmd)); err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to write to stdin: %v", err)
+	}
+	// Send exit to shell
+	if _, err = io.WriteString(stdin, "exit\n"); err != nil {
+		return nil, fmt.Errorf("RunCmdPtySSH: Unable to write to stdin: %v", err)
+	}
+
+	return io.ReadAll(stdout)
+}
+
+// SCP nowadays uses sftp subsystem with no need for scp binary on the target, so use it directly
```

Comment:
Not a blocking comment ... just wanted to say this is AWESOME 🚀 🙂

Reply:
Yeah, tests needed some creative approach when it's about ssh & passwords...